modules/doctor/misc.go

```diff
@@ -189,6 +189,75 @@ func checkDaemonExport(ctx context.Context, logger log.Logger, autofix bool) err
 	return nil
 }
 
+func checkCommitGraph(ctx context.Context, logger log.Logger, autofix bool) error {
+	if err := git.InitOnceWithSync(ctx); err != nil {
+		return err
+	}
+
+	numRepos := 0
+	numNeedUpdate := 0
+	numWritten := 0
+	if err := iterateRepositories(ctx, func(repo *repo_model.Repository) error {
+		numRepos++
+
+		commitGraphExists := func() (bool, error) {
+			// Check commit-graph exists
+			commitGraphFile := path.Join(repo.RepoPath(), `objects/info/commit-graph`)
+			isExist, err := util.IsExist(commitGraphFile)
+			if err != nil {
+				logger.Error("Unable to check if %s exists. Error: %v", commitGraphFile, err)
+				return false, err
+			}
+
+			if !isExist {
+				commitGraphsDir := path.Join(repo.RepoPath(), `objects/info/commit-graphs`)
+				isExist, err = util.IsExist(commitGraphsDir)
```

Comment:
Isn't there a case missing that this directory is empty?
Or what is the file format inside this directory?
Shouldn't this be checked as well?

Reply:
I don't think its worth checking for the format of the directory and/or file at this point.

If you are really interested in how commit graphs are created you are welcome to peruse:

https://github.com/git/git/blob/master/Documentation/technical/commit-graph.txt

and

https://github.com/git/git/blob/master/Documentation/technical/commit-graph-format.txt

Here is the salient points for the contents of commit-graphs

```
A commit-graph chain uses multiple files, and we use a fixed naming convention
to organize these files. Each commit-graph file has a name
`$OBJDIR/info/commit-graphs/graph-{hash}.graph` where `{hash}` is the hex-
valued hash stored in the footer of that file (which is a hash of the file's
contents before that hash). For a chain of commit-graph files, a plain-text
file at `$OBJDIR/info/commit-graphs/commit-graph-chain` contains the
hashes for the files in order from "lowest" to "highest".
```

We could assert that the chain file exists instead of the directory I guess.